elements/lisk-transactions/src/0_transfer_transaction.ts

```diff
@@ -56,13 +74,58 @@ export class TransferTransaction extends BaseTransaction {
 			? rawTransaction
 			: {}) as Partial<TransactionJSON>;
 		// Initializes to empty object if it doesn't exist
-		this.asset = (tx.asset || {}) as TransferAsset;
+		if (tx.asset) {
+			const rawAsset = tx.asset as RawAsset;
+			this.asset = {
+				data: rawAsset.data,
+				recipientId: rawAsset.recipientId,
+				amount: new BigNum(
+					isValidNumber(rawAsset.amount) ? rawAsset.amount : '0',
+				),
+			};
+		} else {
+			// tslint:disable-next-line no-object-literal-type-assertion
+			this.asset = {} as TransferAsset;
+		}
 	}
 
-	protected assetToBytes(): Buffer {
-		const { data } = this.asset;
+	protected getBasicBytes(): Buffer {
```

Comment:
Maybe add comment here about the reasons to override this method? Of course it can be deducted from looking at the code but I think a comment here can speed up any interaction with this code.